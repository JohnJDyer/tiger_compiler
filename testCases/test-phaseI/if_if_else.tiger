let
in
  if 1+2*3 then
    if 1=2*3 then
      flush();
    else
      flush();
    endif;
  endif;
end