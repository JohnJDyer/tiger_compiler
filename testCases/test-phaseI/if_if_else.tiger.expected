main:
	mult, 2, 3, $t0
	add, 1, $t0, $t1
	breq, $t1, 0, if_label0
	mult, 2, 3, $t2
	assign, $t3, 1, 
	breq, 1, $t2, compare_label1
	assign, $t3, 0, 
compare_label1:
	breq, $t3, 0, if_label2
	call, flush
	goto, if_label3, , 
if_label2:
	call, flush
if_label3:
if_label0:
	return, , , 
